File: algorithms/1347.minimum-number-of-steps-to-make-two-strings-anagram/solution.cpp
#include <string>

using namespace std;

class Solution {
public:
    int minSteps(string s, string t) {
        int count[26] = {0};
        int n = s.size();
        for (int i = 0; i < n; i++) {
            count[s[i] - 'a']++;
            count[t[i] - 'a']--;
        }

        int res = 0;
        for (int i = 0; i < 26; i++) {
            if (0 < count[i]) {
                res += count[i];
            }
        }

        return res;
    }
};

File: algorithms/1347.minimum-number-of-steps-to-make-two-strings-anagram/solution.rs
use std::collections::HashMap;

impl Solution {
    pub fn min_steps(s: String, t: String) -> i32 {
        let mut map: HashMap<char, i32> = HashMap::new();
        for c in s.chars() {
            *map.entry(c).or_default() += 1;
        }
        for c in t.chars() {
            *map.entry(c).or_default() -= 1;
        }
        map.values().filter(|&v| *v > 0).sum()
    }

